### PR TITLE
Sub tasks

### LINKED ISSUE #78: Add steps to tasks
Add a list of sub-tasks inside the details pane of a task, it should have an add button and a list of sub-tasks.

### PR DESCRIPTION
**Pull request by edfloreshz**
This PR closes #78 

image

- Added sub-tasks to the task details pane
- The task entry is now using adw::EntryRow

### CODE REVIEW

File: src/widgets/details/factory.rs [hunk 1]
```diff
@@ -337,8 +353,19 @@ impl AsyncFactoryComponent for TaskDetailsFactoryModel {
...
+				adw::PreferencesGroup::default(),
+				sender.input_sender(),
+			),
 			dirty: false,
+		};
+		for sub_task in init.task.sub_tasks {
+			model
+				.sub_tasks
+				.guard()
+				.push_back(SubTaskInit::new(sub_task));
```
Comment (AaronErhardt) [inline src/widgets/details/factory.rs:366]: It's more efficient to get the guard once instead of creating it and dropping it every iteration.

**Review by AaronErhardt — CHANGES_REQUESTED** [review]
Looks good in general, just one minor thing to improve.

File: src/widgets/details/factory.rs (hunk 1 above)
Comment (edfloreshz) [reply src/widgets/details/factory.rs:366]: Right as always! 😉

**Comment by edfloreshz** [comment]
Thanks for the suggestion! It should be fixed now.

**Pull request merged**